bson/src/test/unit/org/bson/BasicBSONEncoderSpecification.groovy

```diff
@@ -179,4 +188,28 @@ class BasicBSONEncoderSpecification extends Specification {
         1 * buffer.writeCString('a')
         1 * buffer.writeInt32(2)
     }
+
+    def 'should encode UUID according to default uuid representation'() {
+        given:
+        def defaultUuidRepresentation = getDefaultUuidRepresentation()
+        def uuid = new UUID(1, 2)
+        def document = new BasicBSONObject()
+        document.append('u', uuid)
+
+        when:
+        setDefaultUuidRepresentation(uuidRepresentation)
+        def bytes = bsonEncoder.encode(new BasicBSONObject(document))
+        def decodedDocument = new BsonDocumentCodec().decode(new BsonBinaryReader(ByteBuffer.wrap(bytes)),
+                DecoderContext.builder().build())
+
+        then:
+        defaultUuidRepresentation == JAVA_LEGACY
+        decodedDocument.getBinary('u').asUuid(uuidRepresentation) == uuid
+
+        cleanup:
+        setDefaultUuidRepresentation(defaultUuidRepresentation)
```

Comment:
Shouldn't `defaultUuidRepresentation == JAVA_LEGACY` be moved (or copied, depending on how the framework executes this test) in the `cleanup` block? I worry that the current code may not always return the static field to its default `JAVA_LEGACY` value.

Reply:
`defaultUuidRepresentation == JAVA_LEGACY` is an assertion, not an assignment.  It's just saying that the original value of the field is JAVA_LEGACY.  But it's weird to put that assertion in this test.  There should just be a separate test, like:

```
def 'default value of defaultUuidRepresentation is JAVA_LEGACY'() {
    expect:
    getDefaultUuidRepresentation() == JAVA_LEGACY
}
```

Reply:
Sorry, I was clearly blind/stupid yesterday.

Reply:
Not at all.  As I was writing that assertion I knew it was confusing, and was being a bit lazy.